src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc

```diff
@@ -227,31 +251,75 @@ void RCCommandRequest::ProcessAccessResponse(
     if (message[app_mngr::strings::msg_params].keyExists(
             message_params::kAllowed)) {
       is_allowed =
-          message[app_mngr::strings::msg_params][message_params::kAllowed]
+          message[app_mngr::strings::msg_params][message_params::kAllowed][0]
               .asBool();
     }
-    if (is_allowed) {
-      resource_allocation_manager_.ForceAcquireResource(module_type,
-                                                        app->app_id());
-      SetResourceState(module_type, ResourceState::BUSY);
-      Execute();  // run child's logic
-    } else {
-      resource_allocation_manager_.OnDriverDisallowed(module_type,
-                                                      app->app_id());
-      SendResponse(
-          false,
-          mobile_apis::Result::REJECTED,
-          "The resource is in use and the driver disallows this remote "
-          "control RPC");
-    }
+    std::string policy_app_id = app->policy_app_id();
+    const auto mac_address = app->mac_address();
+    std::vector<std::string> module_ids{module_id};
+    std::vector<bool> module_allowed{is_allowed};
+    auto module_consents =
+        RCHelpers::FillModuleConsents(module_type, module_ids, module_allowed);
+    rc_consent_manager_.SaveModuleConsents(
+        policy_app_id, mac_address, module_consents);
+    ProcessConsentResult(is_allowed, module_type, module_id, app->app_id());
   } else {
     std::string response_info;
     GetInfo(message, response_info);
     SendResponse(false, result_code, response_info.c_str());
   }
 }
 
-void RCCommandRequest::SendGetUserConsent(const std::string& module_type) {
+void RCCommandRequest::ProcessConsentResult(const bool is_allowed,
+                                            const std::string& module_type,
+                                            const std::string& module_id,
+                                            const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (is_allowed) {
+    SetResourceState(module_type, ResourceState::BUSY);
+    Execute();  // run child's logic
+  } else {
+    resource_allocation_manager_.OnDriverDisallowed(
+        module_type, module_id, app_id);
+
+    std::stringstream ss;
```

Comment:
@ShobhitAd We don't use the std::string in this place because we can't use it in this way: std::string hello_world = "Hello " + "world" (it are wrong operands for operator+). But we need to create the formatted string with module_id and module_type as parts of the string.